Makefile

```diff
@@ -35,17 +35,17 @@ $(BUILD_DIR)/%.o: %.c
 	@mkdir -p $(BUILD_DIR)
 	$(CC) $(CFLAGS) $(COVERAGE_CFLAGS) -I. -c $< -o $@
 
-.PHONY: cov clean
+.PHONY: clean
 
 cov: $(BUILD_DIR)/main
 	@echo "$(COLOR_GREEN)============ RUNNING MAIN PROGRAM ============$(COLOR_RESET)"
 	@./$(BUILD_DIR)/main
 	@gcov $(SOURCES) -o $(BUILD_DIR)
 	@mv *.gcov $(BUILD_DIR)
+	@mkdir $(REPORT_DIR)
```

Comment:
**issue** (blocking): If the dir already exists, then what if the user runs `make cov` again? I think the operation `mkdir` will fail.

We should try to achieve that each [Makefile operation is idempotent](https://news.ycombinator.com/item?id=20375197) (and idiot-proof).

Reply:
**suggestion** (non-blocking): We can fix this by using the `-p` flag for `mkdir`. If you have better ways, then please modify [this line](https://github.com/devprabal/small-linklist/blob/main/Makefile#L28) as well.